handler.go

```diff
@@ -536,27 +567,36 @@ func (fm *Frontman) runServiceCheck(check ServiceCheck) (map[string]interface{},
 	case <-done:
 		return results, err
 	case <-time.After(serviceCheckEmergencyTimeout):
-		log.Errorf("serviceCheck: %s got unexpected timeout after %.0fs", check.UUID, serviceCheckEmergencyTimeout.Seconds())
-		return nil, fmt.Errorf("got unexpected timeout")
+		secs := serviceCheckEmergencyTimeout.Seconds()
+		checkLog.Errorf("serviceCheck: got unexpected timeout after %.0fs", secs)
+		return nil, errors.New("got unexpected timeout")
 	}
 }
 
 func (fm *Frontman) processInput(input *Input, resultsChan chan<- Result) {
-	wg := sync.WaitGroup{}
+	wg := new(sync.WaitGroup)
 	startedAt := time.Now()
 	succeed := 0
 
+	seenServiceChecks := make(map[string]struct{}, len(input.ServiceChecks))
+
 	for _, check := range input.ServiceChecks {
+		if check.UUID == "" {
+			log.Errorf("serviceCheck: missing checkUuid key")
+			continue
+		} else if _, ok := seenServiceChecks[check.UUID]; ok {
```

Comment:
Can be simplified by removing the `else`.

Reply:
And what profit does that change bring to the table?

Reply:
It's simpler.
Simpler code is easier to understand.

Reply:
Do you understand the current code? If so, at least two people are understanding already, I don't think there will be a problem for anyone else.

Reply:
Ok, because you and me understand the code, there is no need to make the code simpler?
This is a very interesting approach. I have worked with this codebase for 6+ moth and we had to clean up a lot of bad things. For example many nested `if` statements that no one understood anymore. This is why I ask you to remove unnecessary branches.
The code flow should be as easy as possible. Check for an error case and `return` if there is an error, if not executions continues. Except you can bring up a valid reason why we should that unnecessary `else` I kindly ask you to remove it. 
`I think that...` or `I don't think that...` is not a valid reason. It's an opinion, where I disagree and I'm accountable in the end.

Reply:
@xlab `else` statement is redundant and confusing here.

Reply:
Fixed